addons/godot-next/references/inspector_controls.gd

```diff
@@ -0,0 +1,62 @@
+extends Reference
+class_name InspectorControls
+
+const ADD_ICON = preload("res://addons/godot-next/icons/icon_add.svg")
+
+class DropdownSelector extends HBoxContainer:
+	
+	var button: ToolButton setget, get_button
+	var dropdown: OptionButton setget, get_dropdown
+	
+	func get_button() -> ToolButton:
+		return get_node("button") as ToolButton
+	
+	func get_dropdown() -> OptionButton:
+		return get_node("dropdown") as OptionButton
+	
+	func get_selected_label() -> String:
+		var dropdown := get_dropdown()
+		var index := dropdown.get_selected_id()
+		return dropdown.get_item_text(index)
+	
+	func get_selected_meta():
+		return get_dropdown().get_selected_metadata()
+
+static func new_button(p_label: String, p_object = null, p_callback: String = "") -> Button:
+	var button = Button.new()
+	button.text = p_label
+	button.name = "button"
+	button.connect("pressed", p_object, p_callback)
+	return button
+
+static func new_dropdown(p_elements: Dictionary, p_object = null, p_callback: String = "") -> OptionButton:
+	var dropdown := OptionButton.new()
+	var index = 0
+	for a_label in p_elements:
+		dropdown.add_item(a_label, index)
+		dropdown.set_item_metadata(index, p_elements[a_label])
+		index += 1
+	dropdown.name = "dropdown"
+	dropdown.size_flags_horizontal = HBoxContainer.SIZE_EXPAND_FILL
+	
+	if p_object:
+		dropdown.connect("item_selected", p_object, p_callback, [dropdown])
+	
+	return dropdown
+
+static func new_dropdown_selector(p_elements: Dictionary, p_object = null, p_callback: String = "") -> DropdownSelector:
```

Comment:
A dropdown really *is* a selector, isn't? And isn't this going to be for a class that has a + button attached? In which case, wouldn't DropdownAppender be a more accurate class name? Can you think of any better alternatives, or do you think the meaning of Dropdown vs. DropdownSelector is clear?

Reply:
DropdownAppender sounds good... I'm not fluent in english so I have some difficulty with namings :)